draft-ietf-quic-transport.md

```diff
@@ -1465,35 +1470,220 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+included data matches the data it sent in a previous PATH_CHALLENGE, the server
+is considered reachable from the new address and the PMTU of the new network
+path is considered verified.
+
+A client SHOULD abandon the new address if it does not receive a PATH_RESPONSE
+after sending some number of PATH_CHALLENGE frames and/or after some time has
+passed. Note that the client may be receiving data from the server during this
+period, but not receiving a PATH_RESPONSE may be indicative of a PMTU detection
+failure.
+
+
+#### Responding to a Server's Validation
+
+A client may receive a PATH_CHALLENGE from a server that is validating the
+client's ownership of its address, as described in {{migration-server}}. The
+client MUST respond to this frame by echoing the data from the server's
+PATH_CHALLENGE frame in a PATH_RESPONSE frame.
+
+A client's PATH_RESPONSE MAY be bundled with other frames and does not require
+additional padding.
+
+A client MAY send more than one PATH_RESPONSE in response to a PATH_CHALLENGE,
+as allowed by the congestion controller, to ensure that packet loss does not
+unduly delay or cause failure of the server's validation process.
+
+A client may trigger the server's validation unwittingly when its public address
+changes, such as when a NAT rebinds the client's private address to a new public
+address.  Though the client may be oblivious of such an address change, it MUST
+respond to the consequent validation.  Receiving a PATH_CHALLENGE from the
+server when the client is not actively changing addresses may be indicative of
+other network activity causing its public address to change, or it may be
+indicative of a server choosing to validate a client's address after a period of
+quiescence.
+
+
+### Server Behavior {#migration-server}
+
+The server responds to any probes sent by the client from a new address,
+participates in the client's PMTU verification, and validates that the client
+owns the new address.  The server MAY validate any client's address at any time,
+but it MUST limit the rate at which data is sent to an unvalidated address, and
+it MUST limit the amount of time for which it sends data to an unvalidated
+address.
+
+
+#### Responding to a Client's Probe
+
+When a server receives a PATH_CHALLENGE frame from a client, it MUST generate a
+PATH_RESPONSE frame that contains the same data as the challenge. The server's
+PATH_RESPONSE MUST be sent to the address from which the client's PATH_CHALLENGE
+was received.
+
+The server sends the PATH_RESPONSE in a packet that is padded to no more than
+the size of the client's packet carrying the PATH_CHALLENGE.  The server SHOULD
+pad the packet to the same size as the client's packet, unless the server has
+reasonable assurance of a smaller PMTU.
+
+The server need not retransmit its PATH_RESPONSE, since it is the client's
+responsibility to send more PATH_CHALLENGE frames when a response is not
+received for a prior one.
```

Comment:
Making them not retransmittable risks false measures of PMTU, doesn't it?  It becomes unclear to the other side whether a drop was random packet loss or the result of oversize datagrams.

Reply:
Is there sufficient text for guidance on how to handle a path that reverts to smaller PMTU during normal operation? I imagine a large stream frame in eternal futile retransmission because it isn't split into smaller frames in smaller packets.